### PR TITLE
fixed bug with typechecker for boolean arrays

### PR DESCRIPTION
**Pull request by jonkhler**
Hi @patrick-kidger ,

thanks again for this excellent library!

When using it extensively in my current project, I realized that there is a minor bug / inconsistency when handling boolean jax arrays. - see PR. 

As far as I could see it the modification does not break anything. However, before the following was failing for me

```python
import jax.numpy as jnp
from jaxtyping import Bool, Array

isinstance(jnp.array(1 > 0), Bool[Array, "..."])
```

### CODE REVIEW

File: jaxtyping/array_types.py [hunk 1]
```diff
@@ -88,9 +88,7 @@ def __init__(self, expr, broadcastable):
 
 
 def _check_dims(
-    cls_dims: List[_AbstractDim],
-    obj_shape: Tuple[int],
-    single_memo: Dict[str, int],
+    cls_dims: List[_AbstractDim], obj_shape: Tuple[int], single_memo: Dict[str, int],
```
Comment (patrick-kidger) [inline jaxtyping/array_types.py:91]: Can this unrelated change be reverted?

**Review by patrick-kidger — COMMENTED** [review]
Ah, good catch! Thank you.

Google requires that all contributors sign a CLA, if you're happy to do that?

Alternatively I'd be happy to quickly write in this fix myself, if that's a faff for you.

File: jaxtyping/array_types.py (hunk 1 above)
Comment (jonkhler) [reply jaxtyping/array_types.py:91]: Regarding CLA - thought I have signed that already. Otherwise feel free to do it yourself :)

**Comment by patrick-kidger** [comment]
Closing in favour of #20.
